packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -760,26 +785,35 @@ export class FluidDataStoreRuntime
 	 */
 	public async getGCData(fullGC: boolean = false): Promise<IGarbageCollectionData> {
 		const builder = new GCDataBuilder();
-		// Iterate over each channel context and get their GC data.
-		await Promise.all(
-			Array.from(this.contexts)
-				.filter(([contextId, _]) => {
-					// Get GC data only for attached contexts. Detached contexts are not connected in the GC reference
-					// graph so any references they might have won't be connected as well.
-					return this.isChannelAttached(contextId);
-				})
-				.map(async ([contextId, context]) => {
-					const contextGCData = await context.getGCData(fullGC);
-					// Prefix the child's id to the ids of its GC nodes so they can be identified as belonging to the child.
-					// This also gradually builds the id of each node to be a path from the root.
-					builder.prefixAndAddNodes(contextId, contextGCData.gcNodes);
-				}),
+		await this.visitLocalBoundContextsDuringSummary(
+			async (contextId: string, context: IChannelContext) => {
+				const contextGCData = await context.getGCData(fullGC);
+				// Prefix the child's id to the ids of its GC nodes so they can be identified as belonging to the child.
+				// This also gradually builds the id of each node to be a path from the root.
+				builder.prefixAndAddNodes(contextId, contextGCData.gcNodes);
+			},
 		);
-
 		this.updateGCNodes(builder);
 		return builder.getGCData();
 	}
 
+	/**
+	 * Helper method for preparing to summarize this channel.
+	 * Runs the callback for each bound context to incorporate its data however the caller specifies
+	 */
+	private async visitLocalBoundContextsDuringSummary(
+		visitor: (contextId: string, context: IChannelContext) => Promise<void>,
+	): Promise<void> {
+		for (const [contextId, context] of this.contexts) {
+			const isAttached = this.isChannelAttached(contextId);
+			// We are not expecting local dds! Summary / GC data may not capture local state.
+			assert(isAttached, 0x17f /* "Not expecting detached channels during summarize" */);
+			if (isAttached) {
+				await visitor(contextId, context);
```

Comment:
There's no way a visitor could add more contexts, right?

Reply:
Actually, it could. As we know, application code may run during summaries and it can create new DDSes (or data stores). But we don't want to summarize them anyway. This behavior hasn't changed with this PR right? The code before also wouldn't consider new contexts?